griptape/drivers/prompt_model/sagemaker_falcon_prompt_model_driver.py

```diff
@@ -7,17 +8,13 @@
 
 @define
 class SageMakerFalconPromptModelDriver(BasePromptModelDriver):
-    tokenizer: BaseTokenizer = field(
-        default=Factory(
-            lambda self: HuggingFaceTokenizer(
-                tokenizer=import_optional_dependency("transformers").AutoTokenizer.from_pretrained(
-                    "tiiuae/falcon-40b", model_max_length=self.max_tokens
-                )
-            ),
-            takes_self=True,
-        ),
-        kw_only=True,
-    )
+    @property
+    def tokenizer(self) -> HuggingFaceTokenizer:
+        return HuggingFaceTokenizer(
```

Comment:
This will create a new Tokenizer each time the property is accessed, this can be expensive. You should cache the value in a field called `_tokenizer`.